Exec/hydro_tests/gamma_law_bubble/Problem_Derive.cpp

```diff
@@ -1,99 +1,390 @@
-                        const int* lo, const int* hi,
-                        const int* domain_lo, const int* domain_hi,
-                        const Real* dx, const Real* xlo,
-                        const Real* time, const Real* dt, const int* bcrec, 
-                        const int* level, const int* grid_no)
-    {
+  Real* const pressure = pressure_rv.dataPtr();
+  Real* const density = density_rv.dataPtr();
+  Real* const temp = temp_rv.dataPtr();
+  Real* const eint = eint_rv.dataPtr();
 
-#pragma gpu
-        derpioverp0(der, AMREX_INT_ANYD(der_lo), AMREX_INT_ANYD(der_hi), *nvar,
-                    data, AMREX_INT_ANYD(data_lo), AMREX_INT_ANYD(data_hi), *ncomp,
-                    AMREX_INT_ANYD(lo), AMREX_INT_ANYD(hi),
-                    AMREX_INT_ANYD(domain_lo), AMREX_INT_ANYD(domain_hi),
-                    AMREX_REAL_ANYD(dx), *time);
+  pressure[0] = pres_base;
+  density[0]  = dens_base;
 
-    }
+  // only initialize the first species
+  Real xn[NumSpec];
+  xn[0] = 1.0_rt;
 
+  // compute the pressure scale height (for an isothermal, ideal-gas
+  // atmosphere)
+  Real H = pres_base / dens_base / std::abs(gravity::const_grav);
 
-    void ca_derrhopert(Real* der, const int* der_lo, const int* der_hi, const int* nvar,
-                       const Real* data, const int* data_lo, const int* data_hi, const int* ncomp,
-                       const int* lo, const int* hi,
-                       const int* domain_lo, const int* domain_hi,
-                       const Real* dx, const Real* xlo,
-                       const Real* time, const Real* dt, const int* bcrec, 
-                       const int* level, const int* grid_no)
-    {
+  for (int j = 0; j < npts_1d; j++) {
 
-#pragma gpu
-        derrhopert(der, AMREX_INT_ANYD(der_lo), AMREX_INT_ANYD(der_hi), *nvar,
-                   data, AMREX_INT_ANYD(data_lo), AMREX_INT_ANYD(data_hi), *ncomp,
-                   AMREX_INT_ANYD(lo), AMREX_INT_ANYD(hi),
-                   AMREX_INT_ANYD(domain_lo), AMREX_INT_ANYD(domain_hi),
-                   AMREX_REAL_ANYD(dx), *time);
+    // initial guess
+     temp[j] = T_guess;
 
-    }
+     if (do_isentropic) {
+       Real z = static_cast<Real>(j) * dx[AMREX_SPACEDIM-1];
+       density[j] = dens_base *
+         std::pow((gravity::const_grav * dens_base * (eos_gamma - 1.0_rt) * z/
+                   (eos_gamma*pres_base) + 1.0_rt), 1.0_rt/(eos_gamma - 1.0_rt));
+     } else {
+       Real z = (static_cast<Real>(j) + 0.5_rt) * dx[AMREX_SPACEDIM-1];
+       density[j] = dens_base * std::exp(-z/H);
+     }
+
+     if (j > 0) {
+        pressure[j] = pressure[j-1] -
+          dx[AMREX_SPACEDIM-1] * 0.5_rt *
+          (density[j] + density[j-1]) * std::abs(gravity::const_grav);
+     }
+
+     eos_t eos_state;
 
+     eos_state.rho = density[j];
+     eos_state.T = temp[j];
+     for (int n = 0; n < NumSpec; n++) {
+       eos_state.xn[n] = xn[n];
+     }
```

Comment:
I guess for this problem it's fine to not do naux, but this PR makes me think we really should have a convenience function that fills an eos_state from a state array so that we don't forget it in general.

Reply:
yeah, I think we can do that for filling from a conserved or primitive state, but this is the initial conditions, so it really only makes sense when the runtime parameters are there.